modules/core/src/main/java/org/apache/ignite/internal/processors/query/RunningQueryManager.java

```diff
@@ -102,20 +139,45 @@ public Long register(String qry, GridCacheQueryType qryType, String schemaName,
      * Unregister running query.
      *
      * @param qryId Query id.
-     * @param failed {@code true} In case query was failed.
+     * @param failReason exception that
      */
-    public void unregister(Long qryId, boolean failed) {
+    public void unregister(Long qryId, @Nullable Throwable failReason) {
         if (qryId == null)
             return;
 
+        boolean failed = failReason != null;
+
         GridRunningQueryInfo qry = runs.remove(qryId);
 
+        // TODO review: would 'assert qry != null' be better?
+        if (qry == null)
+            return;
+
         //We need to collect query history only for SQL queries.
-        if (qry != null && isSqlQuery(qry)) {
+        if (isSqlQuery(qry)) {
             qry.runningFuture().onDone();
 
             qryHistTracker.collectMetrics(qry, failed);
         }
+
+        if (!failed)
+            successQrsCnt.increment();
```

Comment:
we need track only SQL queries, so please do it only in cse isSqlQuery method returns true

Reply:
done.